src/main/python/settings.py

```diff
@@ -31,18 +33,20 @@ def _init_ui(self):
         section_layout.addWidget(window_frame)
         self.stack = add_layout(window_frame, STACK)
 
-        self.workspace = SettingsWorkspace()
+
+        self.workspace = SettingsWorkspace(window_frame, self.cxt)
 
         self.stack.addWidget(self.workspace)
 
 
 class SettingsWorkspace(QFrame):
 
-    def __init__(self, *args, **kwargs):
-        super(QFrame, self).__init__(*args, **kwargs)
+    def __init__(self, parent, cxt:ApplicationContext, *args, **kwargs):
```

Comment:
Why does this add `parent` here? Is `parent` a different class than `QFrame`?

Reply:
The first paramater for `QFrame` is its parent widget. 
Previously, when we create widget, we have:
```
self.workspace = SettingsWorkspace()
```
And now, we modified it to be:
```
self.workspace = SettingsWorkspace(window_frame, self.cxt)
```
The tricky part is, as soon as we want to pass some parameter to an object that hierachy QFrame, the first paramter must be the parent widget (Don't know why). It would raise error if we don't follow.